engine/src/main/java/com/datatorrent/stram/StreamingContainerManager.java

```diff
@@ -2068,6 +2068,22 @@ public void updateRecoveryCheckpoints(PTOperator operator, UpdateCheckpointsCont
     if (checkpointGroup.size() > 1) {
       for (OperatorMeta om : checkpointGroup) {
         Collection<PTOperator> operators = plan.getAllOperators(om);
+        // Include unifiers in group if delay operator is present
+        // Do only if group contains a delay operator
```

Comment:
I saw that when delay is part of the group, the unifiers create a separate checkpoint group even though the upstream and downstream operators are in the same group which seems to be incorrect. In other cases, each of the operators form a separate group and the problem does not arise in that case.